src/capi/azext_capi/custom.py

```diff
@@ -592,33 +593,42 @@ def create_workload_cluster(  # pylint: disable=too-many-arguments,too-many-loca
     logger.warning('✓ Workload access configuration written to "%s"', workload_cfg)
 
     # Install CNI
-    calico_manifest = "https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/master/templates/addons/calico.yaml"  # pylint: disable=line-too-long
+    if windows:
+        pass
+        # TODO: implement this Windows-specific setup
+        # kubectl create ns calico-system
+        # kubectl get configmap kubeadm-config --namespace=kube-system -o yaml \
+        # | sed 's/namespace: kube-system/namespace: calico-system/' \
+        # | kubectl create -f -
+
+    file_base = "https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/main"
+    values_file = f"{file_base}/templates/addons/calico/values.yaml"
+    if user_provided_template:
+        if "ipv6" in user_provided_template:
```

Comment:
do we have a more resilient way of checking of ipv6 instead of just looking for the string "ipv6"  in the template? what if the user provides a template called my-template.yaml that is using ipv6 or dual stack?

If we get the cluster CIDRs, we can use that to check for ipv4 vs. ipv6 vs. dual-stack and pass those in

Reply:
> do we have a more resilient way of checking for ipv6...?

We could write some python to parse the template YAML and try to find the networking section. I just took the approach that CAPZ's Tiltfile does.

> If we get the cluster CIDRs, we can use that to check for ipv4 vs. ipv6 vs. dual-stack and pass those in

I was wondering about that. I had implemented it earlier today but scrapped it when I realized the default works, but probably not for ipv6 / dual-stack. I'll add it back.

Reply:
I did that--see the latest commit. I was just wondering how we'd check explicitly for an ipv6 or dual-stack configuration in the cluster definition, but perhaps passing the CIDRs through as we're doing now is sufficient?

Reply:
is this still a todo? looks like we're still using the name of the template to determine IP family to use

Reply:
So we are fetching the CIDR(s) from the `Cluster` resource now. How were you thinking to use that information to determine which values file to supply?

I suppose we could test `cidr0` to see if it's ipv6, and if both `cidr0` and `cidr1` exist and the latter is ipv6, we assume it's dual-stack.

Reply:
Like this?

```python
file_base = "https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/main"
values_file = f"{file_base}/templates/addons/calico/values.yaml"
if cidr0 and not cidr1:
    interface0 = ipaddress.ip_interface(cidr0)
    if interface0.version == 6:
        values_file = f"{file_base}/templates/addons/calico-ipv6/values.yaml"
elif cidr0 and cidr1:
    interface0 = ipaddress.ip_interface(cidr0)
    interface1 = ipaddress.ip_interface(cidr1)
    if interface0.version == 4 and interface1.version == 6:
        values_file = f"{file_base}/templates/addons/calico-dual-stack/values.yaml"
```

Reply:
yes exactly

if there are two cidrs it should always be dual stack (there is no valid configuration with 2 IPv6 or 2 IPv4 CIDR ranges)

Reply:
> if there are two cidrs it should always be dual stack

For dual-stack, would it be ever valid to have `cidr0` be ipv6 and `cidr1` be ipv4? Just curious. The calico-dual-stack/values.yaml file expects them in the opposite order, so I made the code specific to that.

Reply:
That's a good question. I guess that could be technically possible. I just checked an it does look like our equivalent go helper in the e2e test is tolerant to that (so it would work no matter the order in Cluster.Spec) https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/cni.go#L99

Reply:
I tested locally with `--template ../cluster-api-provider-azure/templates/cluster-template-dual-stack.yaml` and it looks good.

Reply:
Do you want to try to address the edge case where ipv4 and ipv6 cirds are inversed in a custom template?

Reply:
We could do that, but the dual-stack values.yaml file assumes they're in (ipv4, ipv6) order because it specifies different `blockSize` settings. So to support (ipv6, ipv4) properly I think we'd need a different values.yaml file.

Reply:
not necessarily, check out what we're doing here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/cni.go#L99

essentially, we range through Cluster.Spec CIDRs, identify the ipv4 one and the ipv6 one and assign each to a variable, then that lets us put them in the values.yaml file in the right order.